src/calls/components/SingleCallsView.jsx

```diff
@@ -1,12 +1,546 @@
+        // Call
+        callNo: 0,
+        contactType: 'Vrsta kontakta',
+        callDate: '',
+        callDay: 0,
+        callTime: 0,
+        callDur: 0,
+        callType: 'Vrsta poziva',
+
+        // Caller
+        name: '',
+        gender: 'Pol',
+        age: 'Starost',
+        maritalStatus: 'Bracno stanje',
+        numOfCall: 'Koji put zove',
+        planInvolvement: 'Ukljucenost u plan',
+        volunteer: 'Volonter',
+
+        // Call Desc
+        problemType: 'Vrsta problema',
+        suicideRisk: 'Suicidni rizik',
+        suicidFactor: 'Suicidni faktor',
+        lastSuicidTry: 'Prethodni pokusaji suicida',
+        shortContent: '',
+        note: ''
+    };
+
+    // Call
+    handleChangeCallNo = event => {
+        this.setState({ callNo: event.target.value });
+    };
+
+    handleChangeContactType = event => {
+        this.setState({ contactType: event.target.textContent });
+    };
+
+    handleChangeCallDate = event => {
+        this.setState({ callDate: event.target.value });
+    };
+
+    handleChangeCallDay = event => {
+        this.setState({ callDay: event.target.value });
+    };
+
+    handleChangeCallTime = event => {
+        this.setState({ callTime: event.target.value });
+    };
+
+    handleChangeCallDur = event => {
+        this.setState({ callDur: event.target.value });
+    };
+
+    handleChangeCallType = event => {
+        this.setState({ callType: event.target.textContent });
+    };
+
+    // Caller
+    handleChangeName = event => {
+        this.setState({ name: event.target.value });
+    };
+
+    handleChangeGender = event => {
+        this.setState({ gender: event.target.textContent });
+    };
+
+    handleChangeAge = event => {
+        this.setState({ age: event.target.textContent });
+    };
+
+    handleChangeMaritalStatus = event => {
+        this.setState({ maritalStatus: event.target.textContent });
+    };
+
+    handleChangeNumOfCall = event => {
+        this.setState({ numOfCall: event.target.textContent });
+    };
+
+    handleChangePlanInvolvement = event => {
+        this.setState({ planInvolvement: event.target.textContent });
+    };
+
+    handleChangeVolunteer = event => {
+        this.setState({ volunteer: event.target.textContent });
+    };
+
+    // Call Desc
+    handleChangeProblemType = event => {
+        this.setState({ problemType: event.target.textContent });
+    };
+
+    handleChangeSuicidRisk = event => {
+        this.setState({ suicideRisk: event.target.textContent });
+    };
+
+    handleChangeSuicidFactor = event => {
+        this.setState({ suicidFactor: event.target.textContent });
+    };
+
+    handleChangeLastSuicidTry = event => {
+        this.setState({ lastSuicidTry: event.target.textContent });
+    };
+
+    handleChangeShortContent = event => {
+        this.setState({ shortContent: event.target.value });
+    };
+
+    handleChangeNote = event => {
+        this.setState({ note: event.target.value });
+    };
```

Comment:
It's possible to handle all of these input changes with one function, avoiding repetition. [This](https://reactjs.org/docs/forms.html#handling-multiple-inputs) is the relevant bit in React's documentation.

Another way would be to encapsulate input props and their change handlers via hooks. Have a look at [this conference talk](https://www.youtube.com/watch?v=dpw9EHDh2bM) as an introduction to hooks. Dan's section about input fields starts at 17:40.

You don't _have_ to refactor this now if the form works, but the refactor should happen at some point in the near future.

Reply:
Ok, I will try to figure it out later. I fixed package.json

Reply:
Ok, I figured out a logic, 
I will surely fix it in future.